nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -205,25 +205,43 @@ def __init__(
         else:
             self.out_proj = None
             self._feat_out = d_model
+        self.set_pos_emb_max_len(pos_emb_max_len)
+
+    def set_pos_emb_max_len(self, pos_emb_max_len):
+        """ Sets maximum input length.
+            Pre-calculates internal seq_range mask.
+        """
+        self.pos_emb_max_len = pos_emb_max_len
+        device = next(self.parameters()).device
+        seq_range = torch.arange(0, self.pos_emb_max_len, device=device)
+        self.register_buffer('seq_range', seq_range, persistent=False)
 
     @typecheck()
     def forward(self, audio_signal, length=None):
+        audio_size: int = audio_signal.size(-1)
+        if audio_size > self.pos_emb_max_len:
+            logging.warning(
+                f"Audio Signal is too long : {audio_size}, please increase pos_emb_max_len (={pos_emb_max_len}) config parameter!"
+            )
+
         if length is None:
-            length = torch.tensor(audio_signal.size(-1)).repeat(audio_signal.size(0)).to(audio_signal)
+            length = audio_signal.new_full(
+                audio_signal.size(0), audio_size, dtype=torch.int32, device=self.seq_range.device
+            )
+
         audio_signal = torch.transpose(audio_signal, 1, 2)
 
         if isinstance(self.pre_encode, ConvSubsampling):
             audio_signal, length = self.pre_encode(audio_signal, length)
         else:
             audio_signal = self.pre_encode(audio_signal)
-
         audio_signal, pos_emb = self.pos_enc(audio_signal)
-        bs, xmax, idim = audio_signal.size()
-
+        # adjust size
+        audio_size = audio_signal.size(1)
```

Comment:
Can we rename audio_size to max_audio_length?

Reply:
Sure.